x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/timeseriesexplorer_utils.js

```diff
@@ -166,11 +166,7 @@ export function processDataForFocusAnomalies(
         if (record.actual !== undefined) {
           // If cannot match chart point for anomaly time
           // substitute the value with the record's actual so it won't plot as null/0
-          if (chartPoint.value === null) {
-            chartPoint.value = record.actual;
-          }
-
-          if (record.function === ML_JOB_AGGREGATION.METRIC) {
+          if (chartPoint.value === null || record.function === ML_JOB_AGGREGATION.METRIC) {
```

Comment:
This extra check looks good, but testing against the same use case where the check for `if (chartPoint.value === null) ` was introduced - see https://github.com/elastic/kibana/pull/66071, it seems that it isn't actually necessary. The d3 scale seems to work equally well whether the value is supplied as a single number or as an array containing a single item - see https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js#L177.

I'd suggest going ahead with this change as it means all the values in the array of data supplied to the chart are single values. Without it, it means that values at the times of anomalies where there is no corresponding source data are arrays containing a single value.

Note that this change only affects the use case described in https://github.com/elastic/kibana/issues/63098.